tests/test_official.py

```diff
@@ -264,9 +287,222 @@ def check_defaults_type(ptb_param: inspect.Parameter) -> bool:
+        # it could be a list of objects, so let's check that:
+        objs = _extract_words(tg_param_type)
+        # We want to store both string version of class and the class obj itself. e.g. "InputMedia"
+        # and InputMedia because some annotations might be ForwardRefs.
+        if len(objs) >= 2:  # We have to unionize the objects
+            mapped_type: tuple[Any, ...] = (_unionizer(objs, False), _unionizer(objs, True))
+        else:
+            mapped_type = (
+                getattr(telegram, tg_param_type),  # This will fail if it's not from telegram mod
+                ForwardRef(tg_param_type),
+                tg_param_type,  # for some reason, some annotations are not ForwardRefs, i.e. str.
+            )
+        print("len mapped 0 ", mapped_type)
+    elif len(mapped) == 1:
+        print("len mapped 1 ", mapped)
+        mapped_type = mapped.pop()
+
+    # Resolve nested annotations to get inner types.
+    if (ptb_annotation := list(get_args(ptb_param.annotation))) == []:
+        ptb_annotation = ptb_param.annotation  # if it's not nested, just use the annotation
+
+    print(f"{tg_param_type=} ||| {mapped_type=} ||| {ptb_annotation=}")
+
+    if isinstance(ptb_annotation, list):
+        # Some cleaning:
+        # Remove 'Optional[...]' from the annotation if it's present. We do it this way since: 1)
+        # we already check if argument should be optional or not + type checkers will complain.
+        # 2) we don't want to deal with Optional[..] skewing the equality check of our annotation
+        # with the official API, i.e. Optional[int] == int.
+        if type(None) in ptb_annotation:
+            print(f"found None in {ptb_param.name}")
+            ptb_annotation.remove(type(None))
+
+        # Cleaning done... now let's put it back together.
+        # Join all the annotations back (i.e. Union)
+        ptb_annotation = _unionizer(ptb_annotation, False)
+        print(ptb_annotation, "after unionizer")
+
+        # Last step, we need to use get_origin to get the original type, since using get_args
+        # above will strip that out.
+        wrapped = get_origin(ptb_param.annotation)
+        if wrapped is not None:
+            # collections.abc.Sequence -> typing.Sequence
+            if "collections.abc.Sequence" in str(wrapped):
+                wrapped = Sequence
+            ptb_annotation = wrapped[ptb_annotation]
+            print(wrapped, "after wrapped", ptb_annotation)
+        # We have put back our annotation together after removing the NoneType!
+
+    # Now let's do the checking, starting with "Array of ..." types.
+    if "Array of " in tg_param_type:
+        print("array of ", ptb_param.annotation)
+
+        # For exceptions just check if they contain the annotation
+        array_of_exceptions = {  # We accept more types than the official API
+            "results": "InlineQueryResult",
+            "commands": "BotCommand",
+            "keyboard": "KeyboardButton",
+        }
```

Comment:
This should be configurabe somewhere more visible and IMO we should also specify which types we allow in addition

Reply:
specify how? as a comment? If we also hard code the other types, then we have more maintainance work